service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flexibleresource/ControlledFlexibleResource.java

```diff
@@ -0,0 +1,366 @@
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+public class ControlledFlexibleResource extends ControlledResource {
+  private final String typeNamespace;
+  private final String type;
+  @Nullable private final String data;
+
+  @JsonCreator
+  public ControlledFlexibleResource(
+      @JsonProperty("workspaceId") UUID workspaceId,
+      @JsonProperty("resourceId") UUID resourceId,
+      @JsonProperty("name") String name,
+      @JsonProperty("description") String description,
+      @JsonProperty("cloningInstructions") CloningInstructions cloningInstructions,
+      @JsonProperty("assignedUser") String assignedUser,
+      @JsonProperty("privateResourceState") PrivateResourceState privateResourceState,
+      @JsonProperty("accessScope") AccessScopeType accessScope,
+      @JsonProperty("managedBy") ManagedByType managedBy,
+      @JsonProperty("applicationId") String applicationId,
+      @JsonProperty("resourceLineage") List<ResourceLineageEntry> resourceLineage,
+      @JsonProperty("properties") Map<String, String> properties,
+      @JsonProperty("createdByEmail") String createdByEmail,
+      @JsonProperty("createdDate") OffsetDateTime createdDate,
+      @JsonProperty("lastUpdatedByEmail") String lastUpdatedByEmail,
+      @JsonProperty("lastUpdatedDate") OffsetDateTime lastUpdatedDate,
+      @JsonProperty("region") String region,
+      @JsonProperty("typeNamespace") String typeNamespace,
+      @JsonProperty("type") String type,
+      @Nullable @JsonProperty("data") String data) {
+    super(
+        ControlledResourceFields.builder()
+            .workspaceUuid(workspaceId)
+            .resourceId(resourceId)
+            .name(name)
+            .description(description)
+            .cloningInstructions(cloningInstructions)
+            .assignedUser(assignedUser)
+            .accessScope(accessScope)
+            .managedBy(managedBy)
+            .applicationId(applicationId)
+            .privateResourceState(privateResourceState)
+            .resourceLineage(resourceLineage)
+            .properties(properties)
+            .createdByEmail(createdByEmail)
+            .createdDate(createdDate)
+            .lastUpdatedByEmail(lastUpdatedByEmail)
+            .lastUpdatedDate(lastUpdatedDate)
+            .region(region)
+            .build());
+    this.typeNamespace = typeNamespace;
+    this.type = type;
+    this.data = data;
+    validate();
+  }
+
+  /*
+     // TODO: PF-2512 remove constructor above and enable this constructor
+   @JsonCreator
```

Comment:
@ddietterich correct me if I'm wrong, can we enable this now that the blocking tickets are done?

Reply:
You are right, Yu - good catch! The only reason to have the old constructor is to migrate old flights to the new model. This will never have old flights, so it can exclusively use the new model.

If you look at my branch dd-PF-2512-remove-transition you will see what the end state looks like. You can go directly there.